test/e2e/argo-workflows/agent.yaml

```diff
@@ -546,6 +748,38 @@ spec:
           sleep(2000);
         }
 
+  - name: find-kube-state-metrics
+    activeDeadlineSeconds: 200
+    script:
+      image: mongo:3.6.3
+      command: [mongo, "fake-datadog.default.svc.cluster.local/datadog"]
+      source: |
```

Comment:
I think we could avoid some logic in that part by using the database to find a specific metric.
That would limit the JS logic here.

In your case what do you think of:
```js
db.series.find({metric: "kubernetes_state.deployment.replicas_available", tags: {$all: ["namespace:default", "deployment:fake-datadog"] }, "points.0.1": { $eq: 1} })
```
https://docs.mongodb.com/manual/tutorial/query-array-of-documents/

I guess it's the same for the other cpu metrics. Can you make that change too?

Reply:
If the find query isn't enough, the aggregator verb is very powerful.